specification/network/resource-manager/Microsoft.Network/stable/2023-06-01/bastionHost.json

```diff
@@ -434,6 +434,56 @@
         }
       }
     },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Network/bastionHosts/{bastionHostName}/deleteShareableLinksByToken": {
+      "post": {
+        "tags": [
+          "BastionHosts",
+          "BastionShareableLink"
+        ],
+        "operationId": "DeleteBastionShareableLinkByToken",
+        "description": "Deletes the Bastion Shareable Links for all the tokens specified in the request.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupName"
+          },
+          {
+            "$ref": "#/parameters/BastionHostName"
+          },
+          {
+            "$ref": "./network.json#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "./network.json#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/BastionShareableLinkTokenRequest"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success. The operation deleted the BastionShareableLinks associated with the tokens, if they existed. No return body."
+          },
```

Comment:
Its usual to use 204 when there's no return body.

Given this seems to be implementing a bulk delete, I wonder if it would be a better idea though, to return a bulk response - with more detailed info about what resources were successfully deleted, and which were not, or were not found?

(How are you planning to process partially valid requests?)

Reply:
Yes, it is a bulk delete. However, we don't return result content to users. Users will need to call `GetShareableLinks` endpoint if they want to verify if shareable links are deleted. As I mentioned in the purpose of this PR, this change is for adding back a missing endpoint to the specification. The endpoint itself has been release since 2019 and we don't have plan to change its behavior.

Reply:
Do you just reject the request entirely if any requested item was invalid?

What if they were just already deleted? Will you accept that?

Reply:
We don't reject request when items are already deleted.